src/coordinator/parser/promql/parse_test.go

```diff
@@ -69,3 +71,23 @@ func TestDAGWithFunctionCall(t *testing.T) {
 	assert.Equal(t, edges[0].ParentID, parser.NodeID("0"), "fetch should be the parent")
 	assert.Equal(t, edges[0].ChildID, parser.NodeID("1"), "function expr should be the child")
 }
+
+func TestDAGWithAndOp(t *testing.T) {
+	q := "up and up"
+	p, err := Parse(q)
+	require.NoError(t, err)
+	transforms, edges, err := p.DAG()
+	require.NoError(t, err)
+	assert.Len(t, transforms, 3)
+	assert.Equal(t, transforms[0].Op.OpType(), functions.FetchType)
+	assert.Equal(t, transforms[0].ID, parser.NodeID("0"))
+	assert.Equal(t, transforms[1].Op.OpType(), functions.FetchType)
+	assert.Equal(t, transforms[1].ID, parser.NodeID("1"))
```

Comment:
This shouldn't be `logical.AndType` and the next one `functions.FetchType`?

Reply:
the lowest two are fetch types and the AndType is higher order function since you need to do fetches first before you can do AND